app/src/main/java/org/mozilla/focus/widget/Cursor.java

```diff
@@ -78,11 +92,14 @@ public void stopMoving(Direction direction) {
         activeDirections.remove(direction);
         if (activeDirections.size() == 0) {
             handler.removeCallbacks(tick);
+            cursorHandler.postDelayed(cursorTick, 3000);
             moving = false;
         }
     }
 
     private void move() {
+        if (getAlpha() != 1f) { setAlpha(1f); }
```

Comment:
For correctness, I think we'd also need to cancel the active ViewPropertyAnimator.